sdk/python/lib/pulumi/provider/server.py

```diff
@@ -181,6 +181,7 @@ async def _construct_response(
         self, result: ConstructResult
     ) -> proto.ConstructResponse:
         urn = await pulumi.Output.from_input(result.urn).future()
+        assert urn is not None
```

Comment:
mypy didn't like line 198 where we passed Optional[str] to ConstructResponse. I don't think this can ever actually return None so should be fine to assert here (urns always resolve)